src/systems/force_torque/ForceTorque.cc

```diff
@@ -0,0 +1,347 @@
+                 << "]" << std::endl;
+          return true;
+        }
+
+        SensorJointAndLinks sensorJointLinkEntry;
+        sensorJointLinkEntry.joint = jointEntity;
+        sensorJointLinkEntry.jointParentLink = jointParentLinkEntity;
+        sensorJointLinkEntry.jointChildLink = jointChildLinkEntity;
+        this->sensorJointLinkMap[_entity] = sensorJointLinkEntry;
+
+        this->entitySensorMap.insert(
+            std::make_pair(_entity, std::move(sensor)));
+
+        return true;
+      });
+}
+
+//////////////////////////////////////////////////
+void ForceTorquePrivate::Update(const EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("ForceTorquePrivate::Update");
+  _ecm.Each<components::ForceTorque>(
+      [&](const Entity &_entity, const components::ForceTorque *) -> bool
+      {
+        auto it = this->entitySensorMap.find(_entity);
+        if (it != this->entitySensorMap.end())
+        {
+          auto jointLinkIt = this->sensorJointLinkMap.find(_entity);
+          if (jointLinkIt == this->sensorJointLinkMap.end())
+          {
+            ignerr << "Failed to update Force/Torque Sensor: " << _entity
+                   << ". Associated entities not found." << std::endl;
+            return true;
+          }
+
+          auto jointWrench = _ecm.Component<components::JointTransmittedWrench>(
+              jointLinkIt->second.joint);
+
+          // Notation:
+          // X_WJ: Pose of joint in world
+          // X_WP: Pose of parent link in world
+          // X_WC: Pose of child link in world
+          // X_WS: Pose of sensor in world
+          //
+          const auto X_WP =
+              worldPose(jointLinkIt->second.jointParentLink, _ecm);
+          const auto X_WC = worldPose(jointLinkIt->second.jointChildLink, _ecm);
+          // There appears to be a bug worldPose for computing poses of //joint
+          // and its children, so we do it manually here.
+          const auto X_CJ =
+              _ecm.Component<components::Pose>(jointLinkIt->second.joint)
+                  ->Data();
+          auto X_WJ = X_WC * X_CJ;
+
+          auto X_JS = _ecm.Component<components::Pose>(_entity)->Data();
+          auto X_WS = X_WJ * X_JS;
+          auto X_SP = X_WS.Inverse() * X_WP;
+          auto X_SC = X_WS.Inverse() * X_WC;
+
+          // The joint wrench is computed at the joint frame. We need to
+          // transform it the sensor frame.
+          math::Vector3d force =
+              X_JS.Rot().Inverse() * msgs::Convert(jointWrench->Data().force());
+
+          math::Vector3d torque =
+              X_JS.Rot().Inverse() *
+                  msgs::Convert(jointWrench->Data().torque()) +
+              X_JS.Pos().Cross(force);
+
+          it->second->SetForce(force);
+          it->second->SetTorque(torque);
+          it->second->SetRotationParentInSensor(X_SP.Rot());
+          it->second->SetRotationChildInSensor(X_SC.Rot());
```

Comment:
Is it possible for the sensor pose to change w.r.t. the child link's pose?

Reply:
Good point! I don't it can, so I set the rotation during the creation of the sensor instead of recomputing it here every time. https://github.com/ignitionrobotics/ign-gazebo/pull/977/commits/416449995f450e56b3b12ab9d50a0c0eb298c993